lift/Readme.md

```diff
@@ -6,271 +6,360 @@
-declarative migrations as **automatic migrations**.
+</details>
+<details>
+<summary>UpdateEnum</summary>
 
-### Imperative migration
+**TODO**
 
-In traditional migration systems as the one provided by [Active Record](https://edgeguides.rubyonrails.org/active_record_migrations.html), the actual change is
-instead expressed in an imperative manner:
+</details>
+<details>
+<summary>DeleteEnum</summary>
 
-```ruby
-class ChangeUserName < ActiveRecord::Migration[5.0]
-  def change
-    reversible do |dir|
-      change_table :User do |t|
-        t.string :name, null: false
-      end
-    end
-  end
-end
-```
+**TODO**
 
-With imperative migrations, the user needs to both calculate the "diff" in his head and write the needed actions down, which will result in the new schema.
+</details>
 
-While in Prismas model the declarative datamodel is the source of truth for migrations, Active Record takes the imperative migrations as the source of truth for
-migrations. Therefore one can call the imperative migrations also **manual migrations**.
+### Migration
 
-### Datamodel Snapshots
+A Migration is a grouping of one or more Steps. Migrations run in a transaction if the datasource allows it.
 
-As mentioned in [Declarative migration](#how-does-naming-of-migrations-work), the user adjusts the `datamodel.prisma` file, which results in a new migration
-folder when running `prisma migrate`. In order for the migration engine to know which changes need to be applied, it needs a "copy" of the datamodel. As this
-"copy" describes the datamodel in a specific point in time, we also refer to it as a "snapshot".
+### Hook
 
-### Migration Engine
+A hook is a custom shell script that runs either before or after the migration. Hooks give you full control over your migrations. You can write hooks to:
 
-The migration engine is the heart of the whole migration system. It manages the connection to the database and performs the actual changes in the database
-schema. It is written in Rust and will be remotely controlled from the Prisma CLI written in JavaScript.
+- Ensure the data follows new constraints like `unique` or `non null`.
+- Add specific database primitives like functions in Postgres, which are not yet supported by Prisma.
+- Seed the database with data after a migration has been executed.
 
-### Migration History
+## Architecture
 
-When a migration is running or has been executed successfully, Prisma writes this information into a table or collection in the database. This table is also
-known as the "Migration History". It includes all successful and failed migrations of the specific Prisma Project. The benefits of storing this migration
-history are the following:
+Lift has 2 parts: a **Lift Client** and a **Lift Server**. By default, the Lift Server runs locally as a sidecar process. This makes it easy to get started. As
```

Comment:
Valid to link "sidecar process" to e.g. https://blog.davemdavis.net/2018/03/13/the-sidecar-pattern/ for context / noobs like me?